### ISSUE TITLE
Crash to text page 2 when sending disk over audio

### ISSUE THREAD

**Issue by ryandesign** [ADTPro/adtpro#137]
I'm encountering a crash in ADTPro 2.1.0 which sounds similar to #51 but I'm filing it separately in case there is a different cause.

I'm trying to transfer an ancient 5.25" DOS 3.3 floppy from my unenhanced Apple //e to my Mac over audio connection to rescue some old word processing documents. ADTPro reads the first 40 blocks from the disk quickly, then slowly sends blocks up to 16, at which point the speaker clicks, text page 2 is activated, which fills the screen with random-looking inverse, flashing, and normal characters, and I have to restart the computer.

#51 suggests the problem is a bad disk, and there is definitely something wrong with this floppy: three files on the disk, which appear in the CATALOG, display a "file not found" error when trying to BLOAD them. However the disk drive never makes the characteristic clattering noise of a bad disk and the other files can be BLOADed and VERIFY'd. I reformatted a second floppy which is also ancient but which DOS 3.3's INIT command at least seemed to format successfully. I BSAVEd those files that were readable to that second floppy and tried to transfer that second floppy with ADTPro, and it still crashed the same way when sending block 16. This makes me wonder if it's somehow the contents of the files rather than the condition of the disk that's making ADTPro unhappy, though I don't understand how ADTPro could care about the content of the data it's sending. There might also be a chance that the audio cables or the USB audio interface are to blame; I was using a different Mac with a different audio interface and cables than I normally use.

In addition to trying again with my normal Mac, audio interface, and cables, I'll try using the original word processor on the Apple //e to convert its binary files to text and then transferring a disk with those with ADTPro. Maybe it won't crash if sending text files rather than binary.

**Comment by ryandesign** [comment]
Thanks, that gives me some things to try. I'll let you know what works.

**Comment by Shachar** [comment]
Just came here to say I'm facing the precise same issue. Can't transfer a disk. Get garbled text page 2.

**Comment by david-schmidt** [comment]
Can you share the image you're trying to transfer?

**Comment by Shachar** [comment]
My first instinct was to tell you that, of course I can't, as I can't transfer it to a PC. But then I tried to transfer the ADTpro diskette itself, as written by ADTpro bootstrap procedure, and it failed the same way after transferring 16KB.

I should also point out that the amount of data transferred before the failure changes each time, even with the same image. As such, I highly doubt it's because of the image's content. It may have something to do with the volumes or level of noise, though.

**Comment by david-schmidt** [comment]
Yeah, it depends on the direction you're going as to whether you have an image already or not.  Most folks are going the other way, taking an image from the net and sending it to the Apple.
